src/tetris/scene/playScene.ts

```diff
@@ -67,7 +67,21 @@ export default class PlayScene extends Phaser.Scene {
 
 		const matchStart = this._match.startTime / 1000;
 		const currentTime = this._match.currentServerTime / 1000;
-		const preGame = currentTime < matchStart;		
+		const preGame = currentTime < matchStart;
+
+		// Don't allow losses during preGame
+		if (preGame && this._localPlayerField.fieldState === FieldState.Loss) {
+			this._localPlayerField.reset();
+			this._localPlayerField.fieldState = FieldState.Playing;
+		}
+
+		// Disable profiler during preGame
+		if (preGame && !this._game.profiler.isPaused) {
```

Comment:
That also works. I thought this was nicer since it's only called once per "value switch". It can still be simplified to `if (preGame !== profiler.isPaused) { profiler.isPaused = preGame }`